### ISSUE TITLE
update generator

### ISSUE THREAD

**Issue by newExplore-hash** [clovaai/stargan-v2#16]
when updating the generator, the discriminator parameters should be fixed, but i found that you did not 

this part in solver.py (239-241):
x_fake = nets.generator(x_real, s_trg, masks=masks)
out = nets.discriminator(x_fake, y_trg)
loss_adv = adv_loss(out, 1)

i think maybe the following is right:
x_fake = nets.generator(x_real, s_trg, masks=masks)
with torch.no_grad():
       out = nets.discriminator(x_fake, y_trg)

loss_adv = adv_loss(out, 1)

could tell me this right？ tks

**Comment by denadai2** [comment]
I'm not the author, but there are different optimizers for each net, so each optimizer focuses on one net (e.g. discriminator and generator are sometimes fixed)

**Comment by yunjey** [comment]
@newExplore-hash As @denadai2 mentioned, we zero gradients after every update and use separate optimizers so that the discriminator's parameters are not updated when training the generator and vice versa.

**Comment by yunjey** [comment]
Closing the issue, thinking that the question was answered.
Please reopen the issue as needed :)
